spec/controllers/morning_speech_schedule_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+    end
+
+    context "Render data json" do
+      it{expect(response.body).to eq(calendars.to_json(include: {user: {only: :name}}))}
+    end
+  end
+
+  describe "GET #new" do
+    before {get :new}
+
+    context "Return to success response" do
+      it{expect(response.status).to eq(200)}
+    end
+
+    context "Render new template" do
+      it{expect(response).to render_template(:new)}
+    end
+  end
+
+  describe "POST #create" do
+    describe "Ajax create one schedule" do
+      before{post :create, xhr: true, params: {start: DateTime.now.to_s, end: DateTime.tomorrow.to_s}}
+
+      context "Return to success response" do
+        it{expect(response.status).to eq(200)}
+      end
+
+      context "Create new calendar valid day" do
+        it{expect{post :create, xhr: true, params: {start: DateTime.now.to_s, end: DateTime.tomorrow.to_s}}.to change(Calendar, :count).by(2)}
+      end
+
+      context "Render data json" do
+        it{expect(response.body).to eq(Calendar.last.to_json(include: {user: {only: :name}}))}
+      end
+    end
+
+    describe "Create not ajax, mutil schedule valid day" do
+      before{post :create, params: {start: DateTime.now.to_s, end: DateTime.tomorrow.to_s}}
+
+      context "Return to success response" do
+        it{expect(response.status).to eq(302)}
+      end
+
+      context "Create new calendar" do
+        it{expect{post :create, params: {start: DateTime.now.to_s, end: DateTime.tomorrow.to_s}}.to change(Calendar, :count).by(2)}
+      end
+
+      context "Redirect to index calendar" do
+        it{expect(response).to redirect_to(morning_speech_schedule_index_path)}
+      end
+    end
+  end
+
+  describe "PUT #update" do
+    let!(:calendar){FactoryBot.create :calendar,day: DateTime.now, tag: "speech"}
+
+    describe "Valid attribute" do
+      before{put :update, xhr: true, params: {id: calendar.id, day: DateTime.tomorrow.to_s}}
+
+      context "Return to success response" do
+        it{expect(response.status).to eq(200)}
+      end
+
+      context "Change calendar attribute" do
+        it{expect(calendar.reload.day).to eq(DateTime.tomorrow)}
+      end
+
+      context "Render data json" do
+        it{expect(response.body).to eq(calendar.reload.to_json(include: {user: {only: :name}}))}
+      end
+    end
+
+    describe "Invalid attribute" do
+      before{put :update, xhr: true, params: {id: calendar.id, day: DateTime.yesterday.to_s}}
+
+      context "Change calendar attribute" do
+        it{expect(calendar.reload.day.to_s).not_to eq(DateTime.yesterday.to_s)}
+      end
+
+      context "Render data json" do
+        it{expect(response.body).not_to eq(calendar.reload.to_json(include: {user: {only: :name}}))}
+      end
+    end
+  end
+
+  describe "DELETE #destroy" do
+    let!(:calendar){FactoryBot.create :calendar,tag: "speech"}
+
+    context "Delete calendar" do
+      it{expect{delete :destroy, xhr: true, params: {id: calendar.id}}.to change(Calendar, :count).by(-1)}
```

Comment:
cho delete này ra before bên ngoài context
còn ở đây e expect cái khác đi
ví dụ như 
expect(Calendar.count).to eq bằng bao nhiêu đó: 0